lightningd/peer_htlcs.c

```diff
@@ -1442,6 +1442,20 @@ static bool peer_sending_revocation(struct channel *channel,
 	return true;
 }
 
+struct deferred_commigsig {
+	struct sync_waiter sync;
```

Comment:
Wait what and then we `tal_free` via this field?

I tend to see this as bad style for C.  I know it is valid, but still bad style.

At least make a warning comment that *this* field has to be the first and that future refactorings should not move it from its position as first field, because we are going to `tal_free` via a pointer to this field.